backend/services/users/authentication_service.py

```diff
@@ -141,15 +146,30 @@ def generate_session_token_for_user(osm_id: int):
         serializer = URLSafeTimedSerializer(entropy)
         return serializer.dumps(osm_id)
 
+    # code taken from https://github.com/oauthlib/oauthlib/blob/master/oauthlib/common.py
     @staticmethod
-    def generate_authorize_url(callback):
-        token, secret = osm.generate_request_token(callback)
-        url = f"{osm.expand_url(osm.authorize_url)}?oauth_token={url_quote(token)}"
-
-        # Remove tokens from session. The library creates it.
-        session.pop("osm_oauthtok")
+    def generate_random_state(length=30, chars=UNICODE_ASCII_CHARACTER_SET):
+        """Generates a non-guessable OAuth token
+        OAuth (1 and 2) does not specify the format of tokens except that they
+        should be strings of random characters. Tokens should not be guessable
+        and entropy when generating the random characters is important. Which is
+        why SystemRandom is used instead of the default random.choice method.
+        """
+        rand = SystemRandom()
+        return "".join(rand.choice(chars) for x in range(length))
```

Comment:
Since this must be non-guessable and cryptographically secure, can we use a slightly extended character set using `secrets.token_urlsafe(bytes)` instead of `SystemRandom` ? It is equally secure and we get to specify the strength of the token by bytes of data rather than the length of the string. 

```
def generate_random_state(bytes=32):
    import secrets
    return secrets.token_urlsafe(bytes)
```

Some sample strings:

```
>>> secrets.token_urlsafe(32)
'Xe6Tzp565oVznQjb3h-RrTqV9nXMNzypyNBTQaH4UEA'
>>> secrets.token_urlsafe(32)
'I9D4XI8z8D7QuzAFF0EMtQlN7vmpkhOMj_3eBkAW2E0'
>>> secrets.token_urlsafe(32)
'47p-cFn_HFwGnRRai3ymDDIK7Nk5WWvFzln47NHWe1E'
>>> secrets.token_urlsafe(32)
'BjQXLqR9_3U76Wu3xkc9pdZ9xs6uYtSXotsva2t2RU0'
>>> secrets.token_urlsafe(32)
'pYvDwptqbOXULihumsqUPIzjpdT3E_hB_0DysAvr-qk'
>>> secrets.token_urlsafe(32)
'78E74KJ1lYppbwgfW91rQ_HvyuwaR8V51eZzDxLT8rg'
>>> secrets.token_urlsafe(32)
'vsETdIUcFjNI5XrdTnB57Ge2Nz0jHTRo5Ef7-GIayAo'
```

Reply:
That sounds great.